### PR TITLE
Knj 10127 add extension and remove py2 support

### PR DESCRIPTION
**Pull request by cedel1**
upraveny testy a tox (aby to vůbec nějak fungovalo), odebrána podpora pro Py2 a přidáno zpracování extension

### CODE REVIEW

File: pycsob/utils.py [hunk 1]
```diff
@@ -5,23 +5,15 @@
...
+from urllib.parse import urljoin, quote_plus
 
 from . import conf
 
 
 try:
-    from django.utils.timezone import localtime as now
+    from django.utils import timezone as datetime
 except ImportError:
-    from datetime import now
```
Comment (cedel1) [inline pycsob/utils.py:15]: Je to `datetime.datetime.now`, jak se zdá. Ale `datetime.datetime` není package, takže to nejde importovat jako `from datetime.datetime import now`

File: pycsob/utils.py [hunk 2]
```diff
@@ -76,7 +66,7 @@ def str_or_jsbool(v):
 
 
 def dttm(format_='%Y%m%d%H%M%S'):
-    return now().strftime(format_)
+    return datetime.now().strftime(format_)
```
Comment (cedel1) [inline pycsob/utils.py:69]: Souvisí se změnou importu výše.

File: setup.cfg [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [wheel]
 universal = 1
 
-[pytest]
+[tool:pytest]
```
Comment (cedel1) [inline setup.cfg:4]: To původní už nefunguje.

File: tests_pycsob/test_api.py [hunk 4]
```diff
@@ -19,14 +19,15 @@
 class CsobClientTests(TestCase):
 
     def setUp(self):
+        self.key = open(KEY_PATH).read()
```
Comment (cedel1) [inline tests_pycsob/test_api.py:22]: Asi nejrychlejší způsob, jak testy zprovoznit - CsobClient sice po posledních změnách může přijímat i path, ale funkce z utils nemohou....

File: tests_pycsob/test_api.py [hunk 5]
```diff
@@ -19,14 +19,31 @@
...
         self.c = CsobClient(merchant_id='MERCHANT',
                             base_url='https://gw.cz',
-                            private_key_file=KEY_PATH,
-                            csob_pub_key_file=KEY_PATH)
+                            private_key=KEY_PATH,
+                            csob_pub_key=KEY_PATH)
+
+    def test_client_init_can_take_key_string(self):
```
Comment (cedel1) [inline tests_pycsob/test_api.py:28]: Doplnění testů na různé formy initu clienta.

File: requirements-test.pip [hunk 6]
```diff
@@ -1,3 +1,3 @@
 pytest>=2.9.0
-pytest-cov~=2.2.0
+pytest-cov
```
Comment (cedel1) [inline requirements-test.pip]: Tady byl nějaký zádrhel s kontrolou pluginů pytestu - opravili to, ale v pozdější verzi pytest (a pytest-cov zřejmě blokoval upgrade).

**Comment by rubickcz** [comment]
@cedel1 is is possible to show Travis checks here?

File: pycsob/utils.py (hunk 1 above)
Comment (rubickcz) [reply pycsob/utils.py:15]: I've encountered this as well, strange, is it because of new Python version? I believe it was possible in past.

File: requirements-test.pip (hunk 6 above)
Comment (rubickcz) [reply requirements-test.pip]: Please add exact version, this can cause problems in future.

File: README.rst [hunk 7]
```diff
@@ -1,11 +1,12 @@
 pycsob
 ======
 
-.. image:: https://circleci.com/gh/TwistoPayments/pycsob.svg?style=svg
-   :target: https://circleci.com/gh/TwistoPayments/pycsob
+.. image:: https://travis-ci.org/druids/pycsob.svg?branch=master
+    :target: https://travis-ci.org/druids/pycsob
```
Comment (rubickcz) [inline README.rst:5]: Says the repository is inactive.

File: README.rst [hunk 8]
```diff
@@ -1,11 +1,12 @@
...
+
+.. image:: https://codecov.io/gh/druids/pycsob/branch/master/graph/badge.svg
+    :target: https://codecov.io/gh/druids/pycsob
```
Comment (rubickcz) [inline README.rst]: I think we don't have code coverage set up. The link is not working.

File: circle.yml [hunk 9]
```diff
@@ -1,3 +1,3 @@
 machine:
   post:
-    - pyenv global 2.7.12 3.6.2
+    - pyenv global 3.6.12 3.7.9 3.8.5
```
Comment (rubickcz) [inline circle.yml:3]: Is this file needed? We don't use Circle for this lib anymore.

**Review by rubickcz — CHANGES_REQUESTED** [review]

File: circle.yml (hunk 9 above)
Comment (cedel1) [reply circle.yml:3]: Left it from the previous version in case anyone wants to use it.

File: README.rst (hunk 8 above)
Comment (cedel1) [reply README.rst]: Yes, it is not working. Is it possible that it be set up?

File: README.rst (hunk 7 above)
Comment (cedel1) [reply README.rst:5]: Yes, has to be activated after the merge.

File: pycsob/utils.py (hunk 1 above)
Comment (cedel1) [reply pycsob/utils.py:15]: Not sure anymore but I don't think so - if I remember correctly, it failed for all 3 python version. Maybe worked for Py2?

File: README.rst (hunk 7 above)
Comment (cedel1) [reply README.rst:5]: I think that would be necessary for the tests to run in the first place.

File: requirements-test.pip (hunk 6 above)
Comment (cedel1) [reply requirements-test.pip]: This is a test requirement - I don't think it will cause any trouble for production. And if it does cause trouble during tests, then at least we'll know something's wrong.

File: circle.yml (hunk 9 above)
Comment (cedel1) [reply circle.yml:3]: In case it would be merged back to upstream...

File: requirements-test.pip (hunk 6 above)
Comment (cedel1) [reply requirements-test.pip]: Added an approximate version.

File: README.rst (hunk 7 above)
Comment (matllubos) [reply README.rst:5]: image
 now is active

File: pycsob/utils.py (hunk 1 above)
Comment (matllubos) [reply pycsob/utils.py:15]: from datetime import now was nonsence

Comment (matllubos) [reply pycsob/utils.py:15]: ando from datetime.datetime import now was never posible

**Review by matllubos — APPROVED** [review]

**Review by Formulka — APPROVED** [review]

File: README.rst (hunk 7 above)
Comment (rubickcz) [reply README.rst:5]: Can you please amend the commit to see if Travis runs? Or is there anything else to be done to make it working? I still see Travis no output in this PR.

File: README.rst (hunk 8 above)
Comment (rubickcz) [reply README.rst]: I've never used codecov, only coveralls.

File: circle.yml (hunk 9 above)
Comment (rubickcz) [reply circle.yml:3]: I think that's not going to happen, seems it has been abandoned.

File: README.rst (hunk 7 above)
Comment (cedel1) [reply README.rst:5]: I think it needs to be merged first to see it working here. But you can check the PR from my repository, it shows the checks there (I have it set up on my repository).

Comment (rubickcz) [reply README.rst:5]: OK, true.

File: README.rst (hunk 8 above)
Comment (rubickcz) [reply README.rst]: Please either make it working (or try coveralls) or remove it.

**Comment by rubickcz** [comment]
Plese resolve the codecov.io issue (see my comment). Otherwise approved.

**Review by rubickcz — APPROVED** [review]

File: README.rst (hunk 8 above)
Comment (cedel1) [reply README.rst]: Done.

**Comment by rubickcz** [comment]
I'll merge but we need to fix the tests. See https://github.com/druids/pycsob/issues/5
